sympy/parsing/sympy_parser.py

```diff
@@ -626,16 +620,98 @@ def convert_xor(tokens, local_dict, global_dict):
 
     return result
 
+def repeated_decimals(tokens, local_dict, global_dict):
+    """
+    Allows 0.2[1] notation to represent the repeated decimal 0.2111... (19/90)
+
+    Run this before auto_number.
+
+    """
+    result = []
+
+    # num will running match any DECIMAL [ INTEGER ]
+    num = []
+    for toknum, tokval in tokens:
+        if toknum == NUMBER:
+            if (not num and '.' in tokval and 'e' not in tokval.lower() and
+                'j' not in tokval.lower()):
+                num.append((toknum, tokval))
+            elif tokval.isdigit() and len(num) == 2:
+                num.append((toknum, tokval))
+            elif tokval.isdigit() and len(num) == 3 and num[-1][1].isdigit():
+                # Python 3 tokenizes 00123 as '00', '123'
+                # Python 3 tokenizes 01289 as '012', '89'
+                num.append((toknum, tokval))
+            else:
+                num = []
+        elif toknum == OP:
+            if tokval == '[' and len(num) == 1:
+                num.append((OP, tokval))
+            elif tokval == ']' and len(num) >= 3:
+                num.append((OP, tokval))
+            elif tokval == '.' and not num:
+                # handle .[1]
+                num.append((NUMBER, '0.'))
+            else:
+                num = []
+        else:
+            num = []
+
+        result.append((toknum, tokval))
+
+        if num and num[-1][1] == ']':
+            # pre.post[repetend] = a + b/c + d/e where a = pre, b/c = post,
+            # and d/e = repetend
+            result = result[:-len(num)]
+            pre, post = num[0][1].split('.')
+            repetend = num[2][1]
+            if len(num) == 5:
+                repetend += num[3][1]
```

Comment:
In case anyone is wondering why the joining is done here instead of [above](https://github.com/sympy/sympy/pull/14085/files#diff-c3f3cb4004234277039032a7201a70a8R644) it's because we still have to account for truncating `result` [here](https://github.com/sympy/sympy/pull/14085/files#diff-c3f3cb4004234277039032a7201a70a8R665). If we simply join and keep `len(num) == 4`, the size of `result` will be off.